### ISSUE TITLE
Build runner buttons in top bar

### ISSUE THREAD

**Issue by jaumard** [flutter/flutter-intellij#3137]
In most flutter project, you need to have code generation, for example JSON parsing, dependency injection, networks call and more... all those basic stuff take advantage of code generation with the command `flutter packages pub run build_runner watch --delete-conflicting-outputs` would be nice to have a button to launch this command and see the status of it. Even two buttons one for `build` and another for `watch`. 

Here is where I imagine those buttons: 
<img width="132" alt="capture d ecran 2019-01-29 a 14 05 42" src="https://user-images.githubusercontent.com/2970449/51910256-fb6d7300-23ce-11e9-960c-8f7084b51c87.png">

Build action: 2 state, idle and running, if an error occurred the `Event Log` panel can be use to show the error
Watch action: 3 state, idle, watching and running, if an error occurred the `Event Log` panel can be use to show the error

I always forget to launch this command manually and something it crash and stop itself and I don't see it. By having it on the top bar I can quickly see the status and launch/relaunch it

**Comment by jaumard** [comment]
We can also imagine to have a checkbox under Configuration for each time I click on Run or Debug it run before a build command to ensure all my generated source are sync :)

**Comment by devoncarew** [comment]
Thanks for the suggestion!

**Comment by stevemessick** [comment]
Can you add this today using the run configuration editor? There's a section toward the bottom that allows you to specify things that should be done before launching the run config. You might be able to add an external tool that launches a shell script to do what you need.

**Comment by jaumard** [comment]
Hum yes if you want to run this before launching the app, but most of the time you don't want to launch the app, you just want to code a bunch of stuff that are then used to generate other stuff on `build_runnner` for now I'm launching the command line on the Terminal tab, but it lack of visual feedback as you don't know when I generate or if it's even still running or not (as sometime watch quit the process).

**Comment by jaumard** [comment]
Not sure it's needed anymore as we now can put builders into the pub spec and flutter manage all of this automatically :). I let you close this if you're agree ;)

**Comment by Thamizhan** [comment]
This is good feature to have in Intellij. 
Anyhow It'll be more helpful If you give any examples or doc links for how to achieve this with pub spec

**Comment by iangregsondev** [comment]
@jaumard @Thamizhan I would be interested in knowing how to automatically run the code generation via the pub spec in intellij 

Any ideas ?

Thanks.

**Comment by DFelten** [comment]
This is a good plugin for build runner buttons in the top bar of IntelliJ: https://github.com/nEdAy/Flutter-Build-Runner-Helper
